Upgrade EUI to v84.0.0

cee-chen opened the pull request:
## Summary

`eui@83.1.0` ⏩ `eui@84.0.0`

---

## [`84.0.0`](https://github.com/elastic/eui/tree/v84.0.0)

- Updated `EuiDualRange`'s `minInputProps` and `maxInputProps` to support passing more props to underlying inputs ([#6902](https://github.com/elastic/eui/pull/6902))
- `EuiFocusTrap` now supports configuring cross-iframe focus trapping via the `crossFrame` prop ([#6908](https://github.com/elastic/eui/pull/6908))

**Bug fixes**

- Fixed `EuiFilterButton` icon display ([#6900](https://github.com/elastic/eui/pull/6900))
- Fixed `EuiCombobox` compressed plain text display ([#6910](https://github.com/elastic/eui/pull/6910))
- Fixed visual appearance of collapse buttons on collapsible `EuiResizablePanel`s ([#6926](https://github.com/elastic/eui/pull/6926))

**Breaking changes**

- `EuiFocusTrap` now defaults to *not* trapping focus across iframes ([#6908](https://github.com/elastic/eui/pull/6908))

cee-chen committed:
Upgrade EUI to v84.0.0

cee-chen committed:
Fix references to removed `.euiAccordion__padding` classes

cee-chen committed:
Update EuiAccordion snapshots

cee-chen committed:
Update HTML snapshots

cee-chen committed:
[Security] Remove mounted snapshots

- mounted snapshots are a huge headache especially with the Emotion wrappers that don't provide meaningful information and cause snapshot churn on style changes

elasticmachine commented:
Pinging @elastic/eui-team (EUI)

ashokaditya reviewed (approved):
Thanks for the update.

patrykkopycinski commented on yarn.lock:
```diff
@@ -24450,6 +24457,18 @@ react-focus-lock@^2.9.2:
+    use-callback-ref "^1.3.0"
+    use-sidecar "^1.1.2"
+
 react-focus-on@^3.7.0:
```
@cee-chen would you mind to bump kibana `react-focus-on` dependency as well, so we have dependencies aligned between Kibana and EUI?

cee-chen replied:
Great catch on this, thank you @patrykkopycinski! Resolved in https://github.com/elastic/kibana/pull/161716/commits/4042a90cb422dc3151df80a7edc5334d2500b3c4.

Just curious, do you mind clarifying why Kibana needs to specify `react-focus-on` as a dependency at all? As far as I can tell there's no imports or calls to it from within Kibana, just from EUI itself.

patrykkopycinski replied:
I think we needed that to resolve some peer dependencies, but now I think we can remove it from Kibana package.json

cee-chen replied:
Huzzah! My favorite kind of cleanup is the kind where we get to delete stuff 🎉

cee-chen replied:
https://github.com/elastic/kibana/pull/161716/commits/9a33389d24895245299cdab4c7f4c935b0b29c12

thomheymann reviewed (approved):
security changes LGTM

ThomThomson reviewed (approved):
Presentation snapshot changes LGTM

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui-v84.0.x

cee-chen committed:
[PR feedback] Sync `react-focus-on` versions

cee-chen committed:
[PR feedback] Remove `react-focus-on` dependency

Dosant reviewed (approved):
snapshots lgtm

yctercero commented:
Hey! Checking it out for security-detection-engine - just see one little thing where one of our components now doesn't take the full parent width. Is there something we need to update here? 

**Before**
<img width="1464" alt="Screenshot 2023-07-13 at 9 20 58 AM" src="https://github.com/elastic/kibana/assets/10927944/550028d9-8647-4156-a195-6428692183ca">

**After**
<img width="1474" alt="Screenshot 2023-07-13 at 9 21 05 AM" src="https://github.com/elastic/kibana/assets/10927944/6c12f0ad-00aa-4b24-89e9-1093958b5ec6">

cee-chen commented:
Thanks for the thorough QA / catching that @yctercero! That's an odd change, as I'm not seeing anything in v84's changelog that would cause it. Let me pull in latest to make sure it's not related to #161592 and debug locally.

cee-chen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-v84.0.x

cee-chen committed:
Fix Security exceptions list accordion usage of Emotion

- the `width: 100%` was the main fix, but using `@emotion/react` and styled components in the same plugin does not work correctly (per the removed comment). Either migrate the whole component or use `@emotion/css` in the interim instead

cee-chen commented:
@yctercero It looks like an internal accordion change in this release is causing the issue - thank you again for catching it. Changing `display: block` to `width: 100%` resolves the problem (a3c185780cfe68b529963941a7f76c092e85cdf0).

While I was there, I also tweaked the component's use of Emotion - the existing comment that noted issues was due to it being incorrect to use both `@emotion/react` and `styled-components` in the same plugin. See https://github.com/elastic/eui/discussions/6828#discussioncomment-6076157 for more information.

yctercero reviewed (approved):
LGTM - thanks so much for the fix!

banderror reviewed (approved):
Detection Rule Management changes LGTM 👍

semd commented on packages/kbn-securitysolution-grouping/src/components/styles.tsx:
```diff
@@ -36,7 +36,7 @@ export const statsContainerCss = css`
 
 export const groupingContainerCss = css`
-  .groupingAccordionForm .euiAccordion__childWrapper .euiAccordion__padding--m {
+  .groupingAccordionForm .euiAccordion__childWrapper .euiAccordion__children {
```
👍

cee-chen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-v84.0.x

cee-chen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-v84.0.x

cee-chen commented:
Whoa, I'm not totally sure why the PR auto-merged with a failing Security Cypress test, apologies all. I assume it's a flake as the previous run passed - Security team, please feel free to ping me if not!
